### DISCUSSION TITLE
RabbitMQ Using Blue-Green Deployment Strategy

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-server#10784] [Questions] [answered]
Hi,

We have a RabbitMQ version 3.11.x with 3 nodes cluster in our current PROD environment. We want to upgrade from 3.11.x to 3.13.x with out downtime. In the documentation provided I saw the Blue-Green Strategy, for this strategy do we need to create a new cluster setup in a different system? Can you please let me know the steps for Blue-Green Strategy.

Thanks,
Pramod

**Comment by michaelklishin** [comment, marked as answer]
The [Blue-Green Deployment guide](https://www.rabbitmq.com/docs/blue-green-upgrade) explains the idea in details. Specifically, right in the first paragraph it says:

```
Blue-green deployment is an upgrade strategy that is based on the idea of setting up a second RabbitMQ cluster (the "green" one) next to the current production cluster (the "blue" one)
```

The idea is to create a new cluster, import existing definitions there and move your applications in a certain manner, using queue federation as a temporary way to transfer messages to the new cluster where the consumers are (they move first).
